torch/_inductor/utils.py

```diff
@@ -471,6 +474,14 @@ def get_kernel_metadata(node_schedule, wrapper):
 
     from_node_dict = collections.defaultdict(list)
     original_aten_dict = collections.defaultdict(list)
+
+    # sort `inductor_nodes` topologically
+    graph = None
+    if len(inductor_nodes):
+        graph = inductor_nodes[0].graph
+        if all(n for n in inductor_nodes if n in graph.nodes):
+            inductor_nodes = [n for n in graph.nodes if n in inductor_nodes]
```

Comment:
This is O(n) on the FX graph nodes, and so this is ultimately quadratic in the number of kernels in the Inductor graph